plugins/frontend/backstage-plugin-argo-cd/src/components/useAppDetails.ts

```diff
@@ -30,12 +30,47 @@ export const useAppDetails = ({
 }) => {
   const api = useApi(argoCDApiRef);
   const errorApi = useApi(errorApiRef);
+  const configApi = useApi(configApiRef);
 
-  const { loading, value, error, retry } = useAsyncRetry(async () => {
+  const { loading, value, error, retry } = useAsyncRetry(async () => {    
+    let argoSearchMethod: boolean;
     try {
-      if (appName) {
+      argoSearchMethod = configApi.getOptionalConfigArray('argocd.appLocatorMethods') ? true : false
```

Comment:
Here's the other location where we should check the `length` of the config array to be > 0